src/decision/location_selection.jl

```diff
@@ -0,0 +1,266 @@
+- `sum_cover` : array of size (number of scenarios * number of sites) containing the summed coral cover for each site selection scenario.
+- `area_to_seed` : area of coral to be seeded at each time step in km^2
+- `aggregation_function` : function which aggregates ranks output into preferred form, e.g `ranks_to_frequencies`, `ranks_to_location_order`.
+- `target_seed_sites` : list of candidate locations for seeding (indices)
+- `target_shade_sites` : list of candidate location to shade (indices)
+
+# Returns
+-`ranks_store` : number of scenarios * sites * 3 (last dimension indicates: site_id, seed rank, shade rank)
+    containing ranks for each scenario run.
+"""
+function run_site_selection(domain::Domain, 
+    scenarios::DataFrame, 
+    sum_cover::NamedDimsArray, 
+    area_to_seed::Float64;
+    target_seed_sites=nothing, 
+    target_shade_sites=nothing)
+
+    ranks_store = NamedDimsArray(
+        zeros(length(domain.site_ids), 2, nrow(scenarios)),
+        sites=1:length(domain.site_ids),
+        intervention=1:2,
+        scenarios=1:nrow(scenarios),
+    )
+
+    dhw_scens = domain.dhw_scens
+    wave_scens = domain.wave_scens
+
+    # Pre-calculate maximum depth to consider
+    target_dhw_scens = unique(scenarios[:, "dhw_scenario"])
+    target_wave_scens = unique(scenarios[:, "wave_scenario"])
+
+    target_site_ids = Int64[]
+    if !isnothing(target_seed_sites)
+        append!(target_site_ids, target_seed_sites)
+    end
+
+    if !isnothing(target_shade_sites)
+        append!(target_site_ids, target_shade_sites)
+    end
+
+    if isnothing(target_seed_sites) && isnothing(target_shade_sites)
+        target_site_ids = collect(1:length(domain.site_ids))
+    end
+
+    for (scen_idx, scen) in enumerate(eachrow(scenarios))
+        depth_criteria = (domain.site_data.depth_med .<= (scen.depth_min .+ scen.depth_offset)) .& (domain.site_data.depth_med .>= scen.depth_min)
+        depth_priority = findall(depth_criteria)
+
+        considered_sites = target_site_ids[findall(in(depth_priority), target_site_ids)]
+        mcda_vars_temp = DMCDA_vars(domain, scen, considered_sites,  sum_cover[scen_idx, :], area_to_seed, 
+            vec((mean(wave_scens[:, :, target_wave_scens], dims=(:timesteps, :scenarios)) .+ std(wave_scens[:, :, target_wave_scens], dims=(:timesteps, :scenarios))) .* 0.5),
+            vec((mean(dhw_scens[:, :, target_dhw_scens], dims=(:timesteps, :scenarios)) .+ std(dhw_scens[:, :, target_dhw_scens], dims=(:timesteps, :scenarios))) .* 0.5),
+            )
+
+        ranks_store(; scenarios=scen_idx, sites=considered_sites) .= _site_selection(
+            domain, mcda_vars_temp, scen.guided
+        )
+    end
+    # Set filtered locations as n_locs+1 for consistency with time dependent ranks
+    ranks_store[ranks_store.==0.0] .= length(domain.site_ids)+1
+
+    return ranks_store
+
+end
+function run_site_selection(domain::Domain, 
```

Comment:
This returns location rankings.

Could this just be `rank_locations()` ?

Incidentally, here the function is `run_site_selection` but elsewhere it is referred to as `run_location_selection`.